Add Buttons to Remove Projects and Approve Applicants

nvenetucci opened the pull request:
This PR aimed to refactor the `MyProjects` accordion into a separate component for the `ActiveProjects` page, but I abandoned the effort due to the complexity of conditional rendering, which made separating the logic difficult.

Key changes made:
- Kept `MyProjects` page but moved link to the sidebar.
- Added an Archived section to the accordion.
- Displayed project statuses (owner, applicant, collaborator) in the accordion.
- Added buttons to potentially remove projects from the accordion.
- Included approve buttons for applicants on the `Project` page.
- Implemented conditional rendering so only the project owner can see the approve buttons.

MyProjects.tsx
<img width="964" alt="Screenshot 2025-03-02 at 8 48 26 PM" src="https://github.com/user-attachments/assets/b44335b9-f7ae-4767-b7e3-6f641fb14248" />
Project.tsx
<img width="1221" alt="Screenshot 2025-03-02 at 9 02 14 PM" src="https://github.com/user-attachments/assets/f0a1154c-cf30-47da-b922-ed6faff67da2" />

nvenetucci committed:
Add Archived section to MyProjects accordion

nvenetucci committed:
Add project status pills and buttons to inactivate projects

nvenetucci committed:
Update accordion filter to prioritize Archived, Owned, Collaborating, and then Applied projects

nvenetucci committed:
Add placeholder handler functions for archive, withdraw, and leave

nvenetucci committed:
Move MyProjects page link to sidebar

nvenetucci committed:
Add Approve button for applicants

nvenetucci committed:
Add Archived status pill to Project page and fix ProjectCard test

daniiltsioma reviewed (changes requested):
Everything looks really nice, here are some suggestions:
* In the My projects page, it is really hard to see the difference between different sections. Can you add different colors to the "Owned", "Applied", etc.? Just the actual accordion togglers. 
* Can you move "Withdraw", "Leave" buttons inside the project page? So that if I want to leave the project, I need to go to its page. I think this will make the UI less cluttered.
* A minor detail – when opening and closing the sidebar, the header's height jumps up and down. This is because of the "Menu" button height. Can you fix that please?

Otherwise, really great work! I love the applicant/collaborator tags, showing partial usernames for the applicants is great, and the conditional rendering is a nice touch as well!

nvenetucci committed:
Lighter accordion toggles and move Withdraw and Leave buttons to Project page

nvenetucci committed:
Attempt to fix Menu button causing Header height change

nvenetucci commented:
I made the accordion toggles a lighter color, hopefully that improves visibility? Do you think we should move the Archive button to the Project page as well?

daniiltsioma commented:
This looks better! And yes, Archive, Delete, and any other buttons would look better on the Project page. They should require an extra step to access.

nvenetucci committed:
Move Archive button to Project page for project owners

nvenetucci committed:
Move Delete button to Project page for archived projects owned by the user

nvenetucci committed:
Remove placeholder handleDelete function from MyProjects.tsx

nvenetucci commented:
I moved the Archive and Delete buttons to the Project page. Additionally, I updated the Delete button to only render for owners of archived projects, as it didn’t make sense to allow applicants or collaborators to delete projects. Now, if an owner wants to fully delete a project, they must first archive it, and then the Delete button will appear.

daniiltsioma reviewed (approved):
Looks good to me!
